hcsshim.go

```diff
@@ -14,6 +14,7 @@ import (
 //go:generate go run mksyscall_windows.go -output zhcsshim.go hcsshim.go
 
 //sys coTaskMemFree(buffer unsafe.Pointer) = ole32.CoTaskMemFree
+//sys SetCurrentThreadCompartmentId(compartmentId uint32) (hr error) = iphlpapi.SetCurrentThreadCompartmentId
```

Comment:
This is not used in the current version now. It will be used once we add support for overlay networks. 

Overlay networking will work by adding gateway adapters in a compartment. This will allow to have two overlay networks with the same subnet. Since adapters belong to different compartments we would be able to create adapters with the same configuration. This is how things work in Azure and MAS. Our DNS server resolvers will listen on these nics in the non default compartments and hence we need this function in libnetwork. You can take a look at https://github.com/msabansal/libnetwork/blob/5a541a950b48c6086e325d6d1b9a2e3ce2f41c2d/network_windows.go for the usage.